api/src/pcapi/core/educational/api/favorites.py

```diff
@@ -20,24 +21,39 @@ def get_redactors_favorite_templates_subset(
 
 
 def add_offer_to_favorite_adage(
-    redactorId: int,
-    offerId: int,
+    redactor_id: int,
+    offer_id: int,
 ) -> educational_models.CollectiveOfferEducationalRedactor:
-    favorite_offer = educational_models.CollectiveOfferEducationalRedactor(
-        educationalRedactorId=redactorId, collectiveOfferId=offerId
-    )
-    repository.save(favorite_offer)
+    try:
+        with repository.transaction():
+            favorite_offer = educational_models.CollectiveOfferEducationalRedactor(
+                educationalRedactorId=redactor_id, collectiveOfferId=offer_id
+            )
+            db.session.add(favorite_offer)
+    except exc.IntegrityError as exception:
+        favorite_offer = educational_models.CollectiveOfferEducationalRedactor.query.filter_by(
+            educationalRedactorId=redactor_id, collectiveOfferTemplateId=offer_id
+        ).one_or_none()
+        raise exception
```

Comment:
Pourquoi `raise` ?
Peut-être souhaitais-tu comme en-dessous :
```suggestion
        if not favorite_offer:
            raise exception
```

Reply:
Ah oui c'était ce que je voulais faire ! Merci :)